ENH added _repr_jpeg_ method for YouTubeVideo object to allow conversion

bkasel opened the pull request:
of the notebook  with Youtube cells to latex and pdf.

bkasel committed:
ENH added _repr_jpeg_ method for YouTubeVideo object to allow conversion
to latex and pdf.

Carreau commented:
No objection to that. 

Do you want to try adding this information to the docstring ?

It seem also that the canonical way to get the thumbnail is actually `http://img.youtube.com/vi/<insert-youtube-video-id-here>/default.jpg`

And that [other variant are availlable](http://stackoverflow.com/questions/2068344/how-do-i-get-a-youtube-video-thumbnail-from-the-youtube-api) to get bigger sizes.

Carreau commented on IPython/lib/display.py:
```diff
@@ -255,8 +255,12 @@ class YouTubeVideo(IFrame):
+        self.id=id
         src = "https://www.youtube.com/embed/{0}".format(id)
         super(YouTubeVideo, self).__init__(src, width, height, **kwargs)
+    def _repr_jpeg_(self):
```
Also if you are doing modification can you pute a blank line in between the 2 functions ?

bkasel committed:
- changed to default jpeg image for YouTubeVideo representation
- added blank line
- added small note to the docstring

bkasel committed:
- Changed the YouTubeVideo jpeg picture to high quality

bkasel commented:
OK, I did the changes. I was afraid the HQ version would be to wide for the documents, but it looks good.

Cheers

juhasch commented:
Would it be possible to return a HTML object, so the Browser loads the image instead of using `urllib` ?
The current implementation requires the notebook server to have a direct internet connection, which might not be the case (e.g. is behind a proxy which requires authentication).

bkasel commented:
I think this is certainly possible, but then I believe there will not be a jpeg representation in the ipynb file which makes the conversion to latex/pdf more cumbersome. The preferred way the YouTubeVideo is displayed is the Youtube IFrame, which works the way you described.

juhasch commented:
Yes, I see. Using an `Image` object should work, but it will use `urllib`, too.
Btw. in `IPython/core/display.py` there is a fallback to `urllib2` for Python2.
Might be sensible to do so here, too.

bkasel commented:
Yes, I will add the fallback.

And I think I will also include the urlopen call in a try/except block to avoid errors when there is no internet connection.

bkasel committed:
- Added a fallback to urllib2 for python2
- included urlopen call in a try/except block to avoid crash
when no internet connection is available

bkasel commented:
Are there any other changes I have to make?

minrk commented:
Thanks
